File: Lab 4/Lab 4 c++/TechLead.cpp

#include "TechLead.h"

TechLead::TechLead(const std::string& nome, int salario, int numeroEquipes, const std::string& linguagemPrincipal, int projetosEmAndamento)
    : Funcionario(nome, salario), gerente(nome, salario, numeroEquipes), desenvolvedor(nome, salario, linguagemPrincipal), projetosEmAndamento(projetosEmAndamento) {}

Gerente TechLead::getGerente() const {
    return gerente;
}

void TechLead::setGerente(const Gerente& gerente) {
    this->gerente = gerente;
}

Desenvolvedor TechLead::getDesenvolvedor() const {
    return desenvolvedor;
}

void TechLead::setDesenvolvedor(const Desenvolvedor& desenvolvedor) {
    this->desenvolvedor = desenvolvedor;
}

int TechLead::getProjetosEmAndamento() const {
    return projetosEmAndamento;
}

void TechLead::setProjetosEmAndamento(int projetosEmAndamento) {
    this->projetosEmAndamento = projetosEmAndamento;
}

int TechLead::bonus() const {
    return gerente.bonus();
}

std::string TechLead::descricaoGerente() const {
    return gerente.descricaoGerente();
}

std::string TechLead::projetosRealizados() const {
    return desenvolvedor.projetosRealizados();
}

std::string TechLead::descricaoDesenvolvedor() const {
    return desenvolvedor.descricaoDesenvolvedor();
}

std::string TechLead::descricao() const {
    return Funcionario::descricao() + gerente.descricaoGerente() + descricaoDesenvolvedor() +
           "\nProjetos em andamento: " + std::to_string(projetosEmAndamento);
}

File: Lab 4/Lab 4 Java/TechLead.java

public class TechLead extends Funcionario {
    private Gerente gerente;
    private Desenvolvedor desenvolvedor;
    private int projetosEmAndamento;

    public TechLead(String nome, int salario, int numeroEquipes, String linguagemPrincipal, int projetosEmAndamento) {
        super(nome, salario);
        this.gerente = new Gerente(nome, salario, numeroEquipes);
        this.desenvolvedor = new Desenvolvedor(nome, salario, linguagemPrincipal);
        this.projetosEmAndamento = projetosEmAndamento;
    }

    @Override
    public String descricao() {
        return super.descricao() + "\n" + gerente.descricao() + "\n" + desenvolvedor.descricao() + "\nProjetos em andamento: " + projetosEmAndamento;
    }
}
